tsk/fs/logical_fs.cpp

```diff
@@ -479,6 +563,83 @@ find_closest_path_match_in_cache(LOGICALFS_INFO *logical_fs_info, TSK_TCHAR *tar
 	return TSK_OK;
 }
 
+/*
+ * Finds closest sibling match for the given folder.
+ * If best_path is not NULL, caller must free.
+ *
+ * This method does not alter cache ages.
+ * 
+ * @param logical_fs_info The logical file system
+ * @param target_path     The full path being searched for
+ * @param parent_path     The parent path that we found in the cache
+ * @param parent_inum     The addr of the parent path
+ * @param best_name       The best match found in the cache - file name only. (NULL if none are found, must be freed by caller otherwise)
+ * @param best_inum       The inum matching the best path found. LOGICAL_INVALID_INUM if none are found.
+ *
+ * @return TSK_ERR if an error occurred, TSK_OK otherwise
+ */
+static TSK_RETVAL_ENUM
+find_closest_sibling_match_in_cache(LOGICALFS_INFO* logical_fs_info, const TSK_TCHAR* target_path, const TSK_TCHAR* parent_path, TSK_INUM_T parent_inum, TSK_TCHAR** best_name, TSK_INUM_T* best_inum) {
+	TSK_IMG_INFO* img_info = logical_fs_info->fs_info.img_info;
+	IMG_LOGICAL_INFO* logical_img_info = (IMG_LOGICAL_INFO*)img_info;
+	tsk_take_lock(&(img_info->cache_lock));
```

Comment:
This is a really complex function. Is there a unit test that you have written to validate that it is behaving as expected?